apps/api/src/domain/models/digital-text/queries/arango-digital-text-query-repository.integration.spec.ts

```diff
@@ -0,0 +1,390 @@
+        });
+    });
+
+    describe(`fetchMany`, () => {
+        const digitalTexts = digitalTextIds.map((id) =>
+            buildTestInstance(DigitalTextViewModel, {
+                id,
+            })
+        );
+
+        beforeEach(async () => {
+            await testQueryRepository.createMany(digitalTexts);
+        });
+
+        it(`should return the expected digital texts`, async () => {
+            const result = await testQueryRepository.fetchMany();
+
+            expect(result).toHaveLength(digitalTexts.length);
+
+            const missingDigitalTexts = digitalTexts.filter(
+                ({ id }) => !result.some((found) => found.id === id)
+            );
+
+            expect(missingDigitalTexts).toEqual([]);
+        });
+    });
+
+    describe(`count`, () => {
+        const digitalTexts = digitalTextIds.map((id) =>
+            buildTestInstance(DigitalTextViewModel, { id })
+        );
+
+        beforeEach(async () => {
+            await testQueryRepository.createMany(digitalTexts);
+        });
+
+        it(`should return the number of digital texts`, async () => {
+            const result = await testQueryRepository.count();
+
+            expect(result).toBe(digitalTexts.length);
+        });
+    });
+
+    describe(`publish`, () => {
+        const digitalTextId = digitalTextIds[0];
+
+        const targetDigitalText = buildTestInstance(DigitalTextViewModel, {
+            id: digitalTextId,
+            isPublished: false,
+        });
+
+        beforeEach(async () => {
+            await testQueryRepository.create(targetDigitalText);
+        });
+
+        it(`should publish the target digital text`, async () => {
+            await testQueryRepository.publish(digitalTextId);
+
+            const updatedView = (await testQueryRepository.fetchById(
+                digitalTextId
+            )) as DigitalTextViewModel;
+
+            expect(updatedView.isPublished).toBe(true);
+        });
+    });
+
+    describe(`tag`, () => {
+        const existingTagLabel = 'plants';
+
+        const existingTag: TagViewModel = {
+            id: buildDummyUuid(90),
+            label: existingTagLabel,
+            name: buildMultilingualTextWithSingleItem(existingTagLabel),
+            // TODO do we want this here?
+            members: [],
+        };
+
+        const newTagId = buildDummyUuid(91);
+
+        const newTagLabel = 'animals';
+
+        // TODO use event sourced setup?
+        const newTag = buildTestInstance(Tag, {
+            id: newTagId,
+            label: newTagLabel,
+        });
+
+        const targetDigitalText = buildTestInstance(DigitalTextViewModel, {
+            tags: [existingTag],
+        });
+
+        beforeEach(async () => {
+            await databaseProvider.getDatabaseForCollection(ArangoCollectionId.tags).clear();
+
+            await databaseProvider.clearViews();
+
+            await testQueryRepository.create(targetDigitalText);
+
+            await databaseProvider
```

Comment:
This set up will change in the near future. I wonder if there's  a test helper we can introduce to ease the transition later.